File: python/1936.py
def fat(n):
    if n == 1:
        return 1
    return n * fat(n-1)

n = int(input())
count = 0
x = 0

for i in range(1, 10):
    v = fat(i)
    if v > n:
        break
    x = i

for i in range(x, 0, -1):
    fatI = fat(i)
    x = int(n / fatI)
    count += x
    n -= x * fatI
    
print(count)

File: lua/1936.lua
local n = tonumber(io.read())

local x = 1
local fat = 1
local maxX = 0
local maxFat = 0

while fat < n do
    maxFat = fat
    maxX = x
    x = x + 1
    fat = fat * x
end

local count = 1
x = maxX
fat = maxFat
n = n - fat

while n ~= 0 do
    if fat > n then
        fat = fat // x
        x = x - 1
    else
        n = n - fat
        count = count + 1
    end
end

print(count)